### PR TITLE
[Merged by Bors] - refactor: make `PadicInt.valuation` `ℕ`-valued

### LINKED ISSUE #277: `PadicInt.valuation` should be `ℕ`-valued
`PadicInt.valuation` is currently `ℤ`-valued, but only takes nonnegative values.

### PR DESCRIPTION
**Pull request by YaelDillies**
It is currently `ℤ`-valued, but always nonnegative.

Also rename the misnamed `norm_eq_pow_val` to `norm_eq_zpow_neg_valuation`

From FLT

Closes https://github.com/ImperialCollegeLondon/FLT/issues/277

---

- [x] depends on: #20032

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/GroupWithZero/Unbundled.lean
Comment (loefflerd) [inline Mathlib/Algebra/Order/GroupWithZero/Unbundled.lean:1]: These aren't specific to p-adics, so maybe should be split off into a separate mini-PR. What is the rationale for tagging the `1 < a` lemmas as `simp`, but not the `0 < a < 1` lemmas?

File: Mathlib/NumberTheory/Padics/PadicIntegers.lean [hunk 1]
```diff
@@ -306,8 +287,8 @@ theorem exists_pow_neg_lt {ε : ℝ} (hε : 0 < ε) : ∃ k : ℕ, (p : ℝ) ^ (
     norm_cast
     apply le_of_lt
     convert Nat.lt_pow_self _ using 1
-    exact hp.1.one_lt
-  · exact mod_cast hp.1.pos
+    exact (Fact.out : p.Prime).one_lt
```
Comment (loefflerd) [inline Mathlib/NumberTheory/Padics/PadicIntegers.lean:290]: Why this change?

**Review by loefflerd — COMMENTED** [review]
This looks good – thanks for the general cleanup work on the `Padics` files. Should we also have a version of the valuation taking values in `WithTop N`, analogous to `Padic.addValuation`? Anyway, that can be for a later PR.

One question: is there a general reason why it's better not to refer to the `Fact` section variable by name? I agree we don't usually give explicit names to instance variables, but `Fact` is obviously a special case, and it loses a few golf points to keep writing `(Fact.out : p.Prime)` in place of `hp.out`.

File: Mathlib/Algebra/Order/GroupWithZero/Unbundled.lean
Comment (YaelDillies) [reply Mathlib/Algebra/Order/GroupWithZero/Unbundled.lean:1]: I've opened #20032. For the simp tagging, I just followed what was already done in the file

File: Mathlib/NumberTheory/Padics/PadicIntegers.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/NumberTheory/Padics/PadicIntegers.lean:290]: It avoided having to refer to the `Fact` instance by name, but I've reverted that

**Review by Ruben-VandeVelde — APPROVED** [review]
Thanks!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge
